iotdb-core/ainode/resources/syncPythonVersion.groovy

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import groovy.toml.TomlSlurper
+
+def currentMavenVersion = project.version as String
+def currentPythonVersion = currentMavenVersion
+if(currentMavenVersion.contains("-SNAPSHOT")) {
+    currentPythonVersion = currentMavenVersion.split("-SNAPSHOT")[0] + ".dev"
+}
+
+def pyprojectFile = new File("pyproject.toml")
```

Comment:
I set this too absolute path because `../pyproject.toml` doesn't work either.

Reply:
Yeah ... there was an issue that only occured when building with the full reactor ... I fixed that. And extended the script quite a bit.